src/main/java/at/hannibal2/skyhanni/data/Mayors.kt

```diff
@@ -1,25 +1,35 @@
 package at.hannibal2.skyhanni.data
 
-enum class Mayors(val mayorName: String, val color: String, val perk: List<Perks>) {
-    AATROX("Aatrox", "§3", listOf(Perks.SLASHED_PRICING, Perks.SLAYER_XP_BUFF, Perks.PATHFINDER)),
-    COLE("Cole", "§e", listOf(Perks.PROSPECTION, Perks.MINING_XP_BUFF, Perks.MINING_FIESTA)),
-    DIANA("Diana", "§2", listOf(Perks.LUCKY, Perks.MYTHOLOGICAL_RITUAL, Perks.PET_XP_BUFF)),
-    DIAZ("Diaz", "§6", listOf(Perks.BARRIER_STREET, Perks.SHOPPING_SPREE)),
-    FINNEGAN("Finnegan", "§c", listOf(Perks.FARMING_SIMULATOR, Perks.PELT_POCALYPSE, Perks.GOATED)),
-    FOXY("Foxy", "§d", listOf(Perks.SWEET_TOOTH, Perks.BENEVOLENCE, Perks.EXTRA_EVENT)),
-    MARINA("Marina", "§b", listOf(Perks.FISHING_XP_BUFF, Perks.LUCK_OF_THE_SEA, Perks.FISHING_FESTIVAL)),
-    PAUL("Paul", "§c", listOf(Perks.MARAUDER, Perks.EZPZ, Perks.BENEDICTION)),
-
-    SCORPIUS("Scorpius", "§d", listOf(Perks.BRIBE, Perks.DARKER_AUCTIONS)),
-    JERRY("Jerry", "§d", listOf(Perks.PERKPOCALYPSE, Perks.STATSPOCALYPSE, Perks.JERRYPOCALYPSE)),
-    DERPY("Derpy", "§d", listOf(Perks.TURBO_MINIONS, Perks.AH_CLOSED, Perks.DOUBLE_MOBS_HP, Perks.MOAR_SKILLZ)),
-
-    UNKNOWN("Unknown", "§c", listOf()),
+import at.hannibal2.skyhanni.data.jsonobjects.local.MayorJson
+
+enum class Mayors(val mayorName: String, val color: String, val perks: MutableList<Perks>) {
+    AATROX("Aatrox", "§3", mutableListOf()),
+    COLE("Cole", "§e", mutableListOf()),
+    DIANA("Diana", "§2", mutableListOf()),
+    DIAZ("Diaz", "§6", mutableListOf()),
+    FINNEGAN("Finnegan", "§c", mutableListOf()),
+    FOXY("Foxy", "§d", mutableListOf()),
+    MARINA("Marina", "§b", mutableListOf()),
+    PAUL("Paul", "§c", mutableListOf()),
+
+    SCORPIUS("Scorpius", "§d", mutableListOf()),
+    JERRY("Jerry", "§d", mutableListOf()),
+    DERPY("Derpy", "§d", mutableListOf()),
+
+    UNKNOWN("Unknown", "§c", mutableListOf()),
     ;
 
 
     companion object {
         fun getMayorFromName(name: String) = entries.firstOrNull { it.mayorName == name } ?: UNKNOWN
+
+        fun getMayorFromName(name: String, perks: ArrayList<MayorJson.Perk>): Mayors {
+            val mayor = getMayorFromName(name)
+            perks.forEach {
+                mayor.perks.add(Perks.valueOf(it.name))
+            }
```

Comment:
with the other changes combined. Make here this:
``` suggestion
            mayor.perks.forEach{ it.isActive = false}
            perks.filter{mayor.perks.contains(it)}.forEach{ it.isActive = true}
```

Reply:
I dont get this, what is it supposed to do
It doesnt set the active perks to the mayor, does it?

![image](https://github.com/hannibal002/SkyHanni/assets/45315647/5e47c3bb-4b12-4844-a4e5-41d7940ae6d9)
The code itself doesnt work aswell
The .contains() compares the list of Perks with the MayorJson.Perk and throws a "Type inference failed. The value of the type parameter T should be mentioned in input types (argument types, receiver type or expected type). Try to specify it explicitly."
the it in it.isActive is MayorJson.Perk aswell

Ive managed to fix it like that, yet I still dont think it updates the active perks
![image](https://github.com/hannibal002/SkyHanni/assets/45315647/2c017f44-da6a-4625-ace7-b59fe19aa2d9)

Reply:
Wow, I missed that there are two Perk classes in play here.
Fixed version:
```Kotlin
mayor.perks.forEach { it.isActive = false }
perks.mapNotNull { perk -> Perks.entries.firstOrNull { it.perkName ==  perk.name } }.filter{mayor.perks.contains(it)}.forEach{ it.isActive = true}
```

Reply:
Aah I also get it now, sick ty!